### PR TITLE
RFC: Multiple login links for access

### LINKED ISSUE #2115: Support multiple URL properties in amp-access
Currently, only a `login` URL is supported by the component. Ideally, multiple (arbitrary) URLs would be supported. One example use case would be to support distinct `login` and `subscribe` URLs. Otherwise, the "return" URL must be hardcoded into the HTML response and will not support returning the user to the AMP viewer. Here is an example configuration:

```json
{
  "authorization": "https://meter-svc.nytimes.com/meter.js?rid=READER_ID&url=CANONICAL_URL&referer=DOCUMENT_REFERRER&peek=true",
  "pingback": "https://meter-svc.nytimes.com/meter.js?rid=READER_ID&url=CANONICAL_URL&referer=DOCUMENT_REFERRER",
  "login": "https://myaccount.nytimes.com/mobile/login/smart/index.html?campaignId=4YLYF&EXIT_URI=RETURN_URL&rid=READER_ID",
  "subscribe": "https://myaccount.nytimes.com/get-started/auth?OC=20000040560&campaignId=4YLYF&return_url=RETURN_URL&rid=READER_ID"
}
```

And example usage:

```html
<!-- ... -->
<a on="tap:amp-access.subscribe">Subscribe Now</a>
<!-- ... -->
<p>Already a subscriber? <a on="tap:amp-access.login">Log in »</a></p>
<!-- ... -->
```

### PR DESCRIPTION
**Pull request by dvoytenko**
Fixes #2115.

No tests yet!!!

I'd like to discuss this extension to the Login spec.

/cc @cramforce @ashwinlimaye @sebastianbenz @andreban

### CODE REVIEW

**Comment by montmanu** [comment]
@dvoytenko so, building on the example in #2115, this would be the resulting config and HTML:

```json
{
  "authorization": "https://meter-svc.nytimes.com/meter.js?rid=READER_ID&url=CANONICAL_URL&referer=DOCUMENT_REFERRER&peek=true",
  "pingback": "https://meter-svc.nytimes.com/meter.js?rid=READER_ID&url=CANONICAL_URL&referer=DOCUMENT_REFERRER",
  "login": {
    "login": "https://myaccount.nytimes.com/mobile/login/smart/index.html?campaignId=4YLYF&EXIT_URI=RETURN_URL&rid=READER_ID",
    "subscribe": "https://myaccount.nytimes.com/get-started/auth?OC=20000040560&campaignId=4YLYF&return_url=RETURN_URL&rid=READER_ID"
  }
}
```

```html
<!-- ... -->
<a on="tap:amp-access.login-subscribe">Subscribe Now</a>
<!-- ... -->
<p>Already a subscriber? <a on="tap:amp-access.login-login">Log in »</a></p>
<!-- ... -->
```

**Comment by dvoytenko** [comment]
@montmanu Correct, this is a proposed way here. Does this solve your needs?

**Comment by montmanu** [comment]
@dvoytenko yes, it does. thanks!

File: extensions/amp-access/0.1/amp-access.js [hunk 1]
```diff
@@ -687,28 +710,32 @@ export class AccessService {
...
+    log.fine(TAG, 'Start login: ', type);
+    assert(this.config_.loginMap[type],
+        'Login URL is not configured: %s', type);
     // Login URL should always be available at this time.
-    const loginUrl = assert(this.loginUrl_, 'Login URL is not ready');
-    this.analyticsEvent_('access-login-started');
+    const loginUrl = assert(this.loginUrlMap_[type],
+        'Login URL is not ready: %s', type);
+
+    const eventBase = 'access-login' + (type ? '-' + type : '');
```
Comment (cramforce) [inline extensions/amp-access/0.1/amp-access.js]: Maybe also fire untyped event always?

File: extensions/amp-access/0.1/amp-access.js [hunk 2]
```diff
@@ -674,9 +696,10 @@ export class AccessService {
    * Runs the Login flow. Returns a promise that is resolved if login succeeds
    * or is rejected if login fails. Login flow is performed as an external
    * 1st party Web dialog. It's goal is to authenticate the reader.
+   * @param {string} type
```
Comment (cramforce) [inline extensions/amp-access/0.1/amp-access.js]: Document that this can be the empty string and that that is a magic value.

**Comment by cramforce** [comment]
LG in general

File: extensions/amp-access/0.1/amp-access.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access.js]: Done

File: extensions/amp-access/0.1/amp-access.js (hunk 2 above)
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access.js]: Done

**Comment by dvoytenko** [comment]
@cramforce All done. PTAL.

**Pull request merged**